py/runtime0.h

```diff
@@ -76,6 +76,12 @@ typedef enum {
     MP_UNARY_OP_HASH, // __hash__; must return a small int
     MP_UNARY_OP_ABS, // __abs__
     MP_UNARY_OP_INT, // __int__
+    #if MICROPY_PY_BUILTINS_FLOAT
```

Comment:
would be best not to make this enum configurable on `MICROPY_PY_BUILTINS_FLOAT`, it doesn't add any code size having it here for non-float builds, keeps this enum list simpler, and also makes sure `MP_UNARY_OP_SIZEOF` has the same enum value regardless of float/complex configuration (which can be important to maintain the ABI)